### PR TITLE
feat: Reports GST Sales and Purchase Register Beta

### PR DESCRIPTION
**Pull request by priyanshshah2442**

### CODE REVIEW

File: india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+                + si_item.cess_amount
+            ).as_("total_amount"),
+        )
+        .where(si.docstatus == 1)
+    )
+    return query
+
+
+def get_query_with_filters(si, query, filters=None):
+    if not filters:
```
Comment (ljain112) [inline india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.py]: .

File: india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+# For license information, please see license.txt
+
+import frappe
+from frappe import _
+from frappe.query_builder.functions import Sum
+from frappe.utils import getdate
+
+
+def execute(filters=None):
+    if not filters:
```
Comment (ljain112) [inline india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.py:12]: .

File: india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        {
+            fieldtype: "Select",
+            fieldname: "invoice_type",
+            label: __("Invoice Type"),
+            options: "\nB2B\nB2C",
+        }
+    ]
+};
+
+async function set_gstin_options(report) {
```
Comment (ljain112) [inline india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.js]: not required

File: india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.py [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+            ),
+            (
+                si_item.taxable_value
+                + si_item.cgst_amount
+                + si_item.sgst_amount
+                + si_item.igst_amount
+                + si_item.cess_amount
+            ).as_("total_amount"),
+        )
+        .where(si.docstatus == 1)
```
Comment (ljain112) [inline india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.py:215]: check for is_opening

File: india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.py [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+from frappe import _
+from frappe.query_builder.functions import Sum
+from frappe.utils import getdate
+
+
+def execute(filters=None):
+    if not filters:
+        return [], []
+
+    filters = frappe._dict(filters)
```
Comment (ljain112) [inline india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.py:14]: validate from and to date

File: india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.py [hunk 6]
```diff
@@ -0,0 +1,517 @@
...
+            si.is_export_with_gst,
+            si.is_return,
+            si.is_debit_note,
+            si.return_against,
+            IfNull(returned_si.base_grand_total, 0).as_("returned_base_grand_total"),
+            si.gst_category,
+            si_item.gst_treatment,
+            (si_item.cgst_rate + si_item.sgst_rate + si_item.igst_rate).as_("gst_rate"),
+        )
+        .where(si.docstatus == 1)
```
Comment (vorasmit) [inline india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.py]: conditions += " AND IFNULL(billing_address_gstin, '') != company_gstin"

File: india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.py [hunk 7]
```diff
@@ -0,0 +1,517 @@
...
+        .where(si.is_opening != "Yes")
```
Comment (vorasmit) [inline india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.py]: default order by posting date and name

File: india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.py [hunk 8]
```diff
@@ -0,0 +1,517 @@
...
+            si.name.as_("invoice_no"),
+            IfNull(si.base_grand_total, si.grand_total).as_("base_grand_total"),
+            si.posting_date,
+            si.place_of_supply,
+            si.is_reverse_charge,
+            si.is_export_with_gst,
+            si.is_return,
+            si.is_debit_note,
+            si.return_against,
+            IfNull(returned_si.base_grand_total, 0).as_("returned_base_grand_total"),
```
Comment (vorasmit) [inline india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.py]: first pref base_rounded_total, next base_grand_total == invoice_total

File: india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.py [hunk 9]
```diff
@@ -0,0 +1,517 @@
...
+        return invoice.company_gstin[:2] != invoice.place_of_supply[:2]
+
+    def is_b2cl_cn_dn(self, invoice):
+        if invoice.return_against:
+            invoice_value = max(
+                invoice.base_grand_total, invoice.returned_base_grand_total
+            )
+            return invoice_value > B2C_LIMIT and self.is_inter_state(invoice)
+
+        return invoice.base_grand_total > B2C_LIMIT and self.is_inter_state(invoice)
```
Comment (vorasmit) [inline india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.py]: duplication

File: india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.py [hunk 10]
```diff
@@ -0,0 +1,517 @@
...
+                and (self.is_export(row) or self.is_b2cl_cn_dn(row))
+            ):
+                filtered_invoices.append(row)
+
+        return filtered_invoices
+
+
+class GSTR1Invoices(GSTR1Sections):
+    def assign_invoice_category(self, invoices):
+        for row in invoices:
```
Comment (vorasmit) [inline india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.py]: nil rated etc

File: india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.js [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+            fieldname: "summary_by",
+            label: __("Summary By"),
+            options: "Summary by HSN\nSummary by Item",
+            default: "Summary by Item"
+        },
+        {
+            fieldtype: "Select",
+            fieldname: "invoice_category",
+            label: __("Invoice Category"),
+            options: "\nNil-Rated\nExempted\nNon-GST\nCredit/Debit Notes Registered (CDNR)\nCredit/Debit Notes Unregistered (CDNUR)\nB2B\nB2C(Large)\nB2C(Small)\nExport Invoice",
```
Comment (vorasmit) [inline india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.js]: tooltip.

File: india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.py [hunk 12]
```diff
@@ -0,0 +1,517 @@
...
+        filtered_invoices = []
+        for invoice in invoices:
+            if self.is_exempted(invoice):
+                filtered_invoices.append(invoice)
+
+        return filtered_invoices
+
+    def get_non_gst_invoices(self, invoices):
+        filtered_invoices = []
+        for invoice in invoices:
```
Comment (vorasmit) [inline india_compliance/gst_india/report/gst_sales_register_beta/gst_sales_register_beta.py]: add sub categories

**Comment by vorasmit** [comment]
Create new PR just for Purchase.

**Pull request closed without merging**